gapis/api/vulkan/vulkan.go

```diff
@@ -291,7 +307,41 @@ func (API) ResolveSynchronization(ctx context.Context, d *sync.Data, c *path.Cap
 			case VkCmdDebugMarkerEndEXTArgsʳ:
 				popMarker(DebugMarker, uint64(i), nCommands)
 			}
+
+			// Handle draw commands grouping.
+			cmdName := ""
+			if generatingId != nil {
+				cmdName = generatingId.CmdName()
+			}
+			isDrawCmd := strings.HasPrefix(cmdName, "vkCmdDraw") || strings.HasPrefix(cmdName, "vkCmdDispatch")
+			isStateSettingCmd := (strings.HasPrefix(cmdName, "vkCmdSet") || strings.HasPrefix(cmdName, "vkCmdBind")) &&
+				!strings.HasPrefix(cmdName, "vkCmdSetEvent")
+			if isStateSettingCmd && canStartDrawGrouping {
+				markerStack = append(markerStack,
+					&markerInfo{
+						name:   "Draw Group", // Shouldn't really get this name. Normally the draw command group's name will be decided later at stack pop time.
+						ty:     DrawGroupMarker,
+						start:  uint64(i),
+						end:    uint64(i),
+						parent: append(api.SubCmdIdx{}, idx...),
+					})
+				canStartDrawGrouping = false
+			} else if isDrawCmd {
+				groupName := cmdName
+				if strings.HasPrefix(groupName, "vkCmd") { // Remove "vkCmd".
+					groupName = groupName[5:len(groupName)]
+				}
+				popMarkerWithNewGroupName(DrawGroupMarker, uint64(i), groupName)
+				canStartDrawGrouping = true
+			} else if !isStateSettingCmd && !isDrawCmd && !canStartDrawGrouping {
+				// Handle an edge case where a group of state setting commands are
+				// followed by something other than a drawing command.
+				groupName := "State Setting Group"
```

Comment:
If we default the group name to this rather than "Draw Group" then maybe this can be simplified a bit ["falling off the end" and just calling the normal popMarker will do the correct thing then?]